core/src/advice/map.rs

```diff
@@ -17,32 +21,47 @@ use crate::utils::{ByteReader, ByteWriter, Deserializable, DeserializationError,
 /// associated with a given key onto the advice stack using `adv.push_mapval` instruction. The VM
 /// can also insert new values into the advice map during execution.
 #[derive(Debug, Clone, Default, PartialEq, Eq)]
-pub struct AdviceMap(BTreeMap<Word, Vec<Felt>>);
+pub struct AdviceMap(BTreeMap<Word, Arc<[Felt]>>);
 
 /// Pair representing a key-value entry in an [`AdviceMap`]
-type MapEntry = (Word, Vec<Felt>);
+type MapEntry = (Word, Arc<[Felt]>);
 
 impl AdviceMap {
     /// Creates a new advice map.
     pub fn new() -> Self {
-        Self(BTreeMap::<Word, Vec<Felt>>::new())
+        Self::default()
     }
 
     /// Returns the values associated with given key.
     pub fn get(&self, key: &Word) -> Option<&[Felt]> {
-        self.0.get(key).map(|v| v.as_slice())
+        self.0.get(key).map(|v| &**v)
+    }
+
+    /// Returns a pointer to the values associated with given key.
+    pub fn get_arc(&self, key: &Word) -> Option<&Arc<[Felt]>> {
```

Comment:
If we're going to "lock" ourselves to exposing the underlying `Arc`, does `get()` provide any value anymore? i.e. should `get()` return `Option<&Arc<[Felt]>>`? `Arc<[Felt]` derefs to `&[Felt]`, so should work without breaking user code?